src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/VisualStudioBrowserRefreshServer.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Net;
+using Microsoft.DotNet.HotReload;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.HotReload;
+
+internal sealed class VisualStudioBrowserRefreshServer(
+    ILogger logger,
+    ILoggerFactory loggerFactory,
+    string projectName,
+    int port,
+    int sslPort,
+    string virtualDirectory)
+    : AbstractBrowserRefreshServer(GetMiddlewareAssemblyPath(), logger, loggerFactory)
```

Comment:
Does this PR also comes with vscode implementation of browser refresh server?

Reply:
Yes. `BrowserRefreshServer` is Kestrel based.